src/formats/XYZ.cpp

```diff
@@ -78,19 +103,396 @@ optional<uint64_t> XYZFormat::forward() {
+                }
+            }
+        }
+        result += fmt::format(" Lattice=\"{:g} {:g} {:g} {:g} {:g} {:g} {:g} {:g} {:g}\"",
+            matrix[0][0], matrix[1][0], matrix[2][0],
+            matrix[0][1], matrix[1][1], matrix[2][1],
+            matrix[0][2], matrix[1][2], matrix[2][2]
+        );
+    }
+
+    auto sorted_properties = std::map<string_view, Property>{frame.properties().begin(), frame.properties().end()};
+    // support for generic frame properties
+    for (const auto& it: sorted_properties) {
+        if (contains_space(it.first)) {
+            warning("Extended XYZ", "frame property '{}' contains spaces, skipping", it.first);
+            continue;
+        }
+
+        result += fmt::format(" {}=", it.first);
+        switch (it.second.kind()) {
+        case Property::STRING:
+            result += fmt::format("\"{}\"", it.second.as_string());
+            break;
+        case Property::BOOL:
+            result += fmt::format("{}", it.second.as_bool() ? 'T' : 'F');
+            break;
+        case Property::DOUBLE:
+            result += fmt::format("{:g}", it.second.as_double());
+            break;
+        case Property::VECTOR3D:
+            {
+                auto v = it.second.as_vector3d();
+                result += fmt::format("\"{:g} {:g} {:g}\"", v[0], v[1], v[2]);
+            }
+            break;
+        }
+    }
+    return result;
+}
+
+/// mapping name=>properties values. The use of string_view is safe since the
+/// original string_view (as returned by file.readline()) will stay alive longer
+/// than this map.
+using extended_xyz_properties_map_t = std::unordered_map<string_view, string_view>;
+
+/// A simple parser for the extended XYZ comment line. This parser bails out as
+/// soons as possible if  the line does not seems to follow extended XYZ
+/// convention.
+///
+/// This line should contains key/values pairs separated by spaces, while the
+/// key and the value are separated by an = sign. If the value contains spaces,
+/// it is then quoted, both single quotes and double quotes are accepted here.
+/// Example: Properties=species:S:1:pos:R:3:bad:I:1 Lattice=" 3 0 0 0 4 0 0 0 5" name='simple file'
+struct extended_xyz_parser {
+public:
+    extended_xyz_parser(string_view line): current_(line.data()), end_(line.data() + line.size()) {}
+    ~extended_xyz_parser() = default;
+    extended_xyz_parser(extended_xyz_parser&&) = delete;
+    extended_xyz_parser(const extended_xyz_parser&) = delete;
+    extended_xyz_parser& operator=(extended_xyz_parser&&) = delete;
+    extended_xyz_parser& operator=(const extended_xyz_parser&) = delete;
+
+    extended_xyz_properties_map_t parse() {
+        extended_xyz_properties_map_t properties;
+        while (!done()) {
+            auto name = get_property_name();
+            if (!name.empty() && current() == '=') {
+                advance();
+            } else {
+                // this does not look like a property, bail out
```

Comment:
Nice catch! This code was written with the assumption that it can also read files not following the extended XYZ convention.

Trying to continue reading could work, the only issue I can see with this is if a file contains `Properties=` in the comment line but is not intended to be an extended XYZ file

Reply:
Didn't think about that. Maybe its ok to bail out, if the line doesn't start with an key/value pair (AFAIK QUIP always starts with the lattice). So one would only have a problem if a non exyz comment line starts with something that looks like a valid key/value pair.

Reply:
Should be working now, I've added such None-valued property in the test files: https://github.com/chemfiles/tests-data/commit/44c73902de2ac14ca9a083b3da6916bc44aba6dc

Reply:
I haven't kept fully track of the discussion on the ase repo, but ase's [newly added documentation](https://gitlab.com/ase/ase/-/blob/master/ase/io/extxyz.py#L702-704) seems to suggest that keys without value should be `true`. I'm currently a little bit lost in the documentations and don't know which one is "right". (So there are no `None` type values? But why handles QUIP such cases as `None` types?)
Chemfiles currently skips keys without value. Should we read such values as `true` instead?

Reply:
I'll double check what ASE & QUIP are doing, but if ASE is setting these to True, I am fine with doing the same =)

Reply:
Actually, nowadays QUIP uses ASE directly to read files instead of there own code, so setting properties without values to true should be enough